content/articles/data-visualization-in-react-using-apexcharts/index.md

```diff
@@ -0,0 +1,260 @@
+        type="bar"
+        width="450"
+      />
+    </div>
+  )
+}
+
+export default MyCharts;
+```
+On the second line of our file, we are importing Chart from `react-apexcharts`. We then declare two variables, `series` and `options`. We will then set the `series` and `options` props in the `<Chart />` component to these variables.  `<Chart />` component accepts the following props;
+
+- `options`: This is a prop type of object. All the optional chart configurations are passed here.  In the `options` variable, the `chart` key has a value of an object `{ id: 'bar-chart'}`, which is the id of our chart. The key `xaxis` has a value of `categories`, which is the label that will be on the x-axis.
+
+- `series`: This is a prop type of an array. It contains the data that we want to display on the y-axis.
+
+- `type`: This is a prop type of string. It defines the type of chart that we want to display. In our case, a `bar` chart. There are other types such as `line` and `donut`.
+
+- `width`: Can be a string or a number. It defines the width of the chart.
+
+- `height`: Can be a string or a number. It defines the height of the chart.
+### Creating fake REST API using JSON server
+This section is for the readers using a JSON server to create the fake REST API. If you have an existing API endpoint, you can skip to the next section.
+Create a `data` folder at the root of your project. Inside this folder, create a `db.json` file which will contain our data. Include the following in the `db.json` file.
+
+#### db.json
+```javascript
+{
+  "temperature": [
+    {
+      "id": 1,
+      "date": "03/01/2022",
+      "average_temp": 15
+    },
+    {
+      "id": 2,
+      "date": "03/02/2022",
+      "average_temp": 27
+    },
+    {
+      "id": 3,
+      "date": "03/03/2022",
+      "average_temp": 18
+    },
+    {
+      "id": 4,
+      "date": "03/04/2022",
+      "average_temp": 20
+    },
+    {
+      "id": 5,
+      "date": "03/05/2022",
+      "average_temp": 23
+    },
+    {
+      "id": 6,
+      "date": "03/06/2022",
+      "average_temp": 17
+    },
+    {
+      "id": 7,
+      "date": "03/07/2022",
+      "average_temp": 15
+    }
+  ]
+}
+```
+This file contains dummy data about the average temperature (in degrees Celsius) in Nairobi City between 1st March 2022 and 7th March 2022.
+
+Open another terminal and run the following command to start the server.
+```bash
+json-server --watch data/db.json --port 8000
+```
+
+The command above starts our JSON server on port 8000. We specify a port number because when you run the JSON server, it will try to run on port 3000, but we are already using this port to run our React application. Thus, we use port 8000. On the terminal, you will see this URL, `http://localhost:8000/temperature`. This will be the API endpoint that we will use to fetch the data in our application.
+
+### Fetching data
+In `charts.js`, we will use the useEffect hook to fetch data that will be displayed in the chart.
```

Comment:
Hooks such as useEffect need to be put inside `` . 

Go through the whole article and edit accordingly.